@xen-orchestra/backups/_cleanVm.js

```diff
@@ -184,7 +184,7 @@ exports.cleanVm = async function cleanVm(vmDir, { remove, merge, onLog = noop })
   await asyncMap(xvas, async path => {
     // check is not good enough to delete the file, the best we can do is report
     // it
-    if (!(await isValidXva(path))) {
+    if (handler._getFilePath !== undefined && !(await isValidXva(handler._getFilePath('/' + path)))) {
```

Comment:
I think, the best answer here would be to make `isValidXva` work with an fs handler.

I gave it a quick look and I think it should be straightforward, what do you think?

Reply:
It can also be added to the `RemoteAdapter`.

Reply:
Sure :+1: